### PR TITLE
fix(PXP-11385): Adds single user google group updates

### PR DESCRIPTION
**Pull request by k-burt-uch**
Fixes issue with passports where a single user google group updates would clear existing membership from the GBAG.

Link to JIRA ticket if there is one: 

### Bug Fixes
- Fixes issue where google group updates for passport sync would clear existing membership from Google Bucket Access Groups.

### CODE REVIEW

File: fence/resources/google/access_utils.py [hunk 1]
```diff
@@ -42,7 +42,83 @@ class GoogleUpdateException(Exception):
...
+
+        if google_update_failures:
+            raise GoogleUpdateException(
+                f"FAILED TO UPDATE GOOGLE GROUPS FOR USER {user_email} (see previous errors)."
+            )
+
+
+def bulk_update_google_groups(google_bulk_mapping, keep_existing=False):
     """
     Update Google Groups based on mapping provided from Group -> Users.
```
Comment (Avantol13) [inline fence/resources/google/access_utils.py:123]: lets add `keep_existing` to the docstring

File: fence/resources/google/access_utils.py [hunk 2]
```diff
@@ -42,7 +42,83 @@ class GoogleUpdateException(Exception):
...
+                gcm.remove_member_from_group(user_email, group)
+            except Exception as exc:
+                logger.error(
+                    f"ERROR: FAILED TO REMOVE USER {user_email} FROM "
+                    f"GOOGLE GROUP {group}! This sync will continue to update other groups. Exc: "
+                    f"{traceback.format_exc()}"
+                )
+                google_update_failures = True
+
+        if google_update_failures:
```
Comment (Avantol13) [inline fence/resources/google/access_utils.py:115]: Thanks for all this error handling and clear logging messages 💯

File: fence/sync/sync_users.py [hunk 3]
```diff
@@ -993,14 +990,14 @@ def sync_to_storage_backend(self, user_project, user_info, sess, expires):
...
             user_project_lowercase,
             sess,
-            google_bulk_mapping=google_bulk_mapping,
+            google_bulk_mapping=google_group_user_mapping,
             expires=expires,
         )
 
         if config["GOOGLE_BULK_UPDATES"]:
-            self.logger.info("Doing bulk Google update...")
-            bulk_update_google_groups(google_bulk_mapping)
```
Comment (Avantol13) [inline fence/sync/sync_users.py:1002]: So this is called elsewhere by usersync, in this function: https://github.com/uc-cdis/fence/blob/e2a43234785ca09eb05aa49771ede5561c477472/fence/sync/sync_users.py#L1627

and `bulk_update_google_groups` still gets called there: https://github.com/uc-cdis/fence/blob/e2a43234785ca09eb05aa49771ede5561c477472/fence/sync/sync_users.py#L934

which means it'll wipe out groups.

Actually, I don't see where `sync_to_storage_backend` is called at all

File: fence/resources/google/access_utils.py (hunk 1 above)
Comment (k-burt-uch) [reply fence/resources/google/access_utils.py:123]: Oh, I need to remove this entirely. The above method should be used in place of that.

Comment (Avantol13) [reply fence/resources/google/access_utils.py:123]: okay that makes a lot more sense actually. There's another place where this is called (see my other comment)

**Review by Avantol13 — APPROVED** [review]

**Pull request merged**